hazelcast/src/main/java/com/hazelcast/util/SampleableConcurrentHashMap.java

```diff
@@ -94,6 +95,43 @@ public int fetch(int tableIndex, int size, List<Data> keys) {
         return nextTableIndex;
     }
 
+    /**
+     * Fetches entries from given <code>tableIndex</code> as <code>size</code>
+     * and puts them into <code>entries</code> list.
+     *
+     * @param tableIndex           Index (checkpoint) for starting point of fetch operation
+     * @param size                 Count of how many entries will be fetched
+     * @param entries              List that fetched entries will be put into
+     * @return the next index (checkpoint) for later fetches
+     */
+    public int fetchEntries(int tableIndex, int size, List<Map.Entry<Data, V>> entries) {
```

Comment:
Seems tha we don't need `Data` type as type of the key. So I think we can use `List<Map.Entry<K, V>> entries` parameter for both of the `fetchEntries` and `fetchKeys` methods.